scripts/tools/GenerateKeys.php

```diff
@@ -0,0 +1,128 @@
+                'longPrefix' => 'key_id',
+                'description' => 'Lti Platform key chain id',
+                'required' => true,
+                'cast' => 'string'
+            ],
+            'key_name' => [
+                'prefix' => 'kn',
+                'longPrefix' => 'key_name',
+                'description' => 'Lti Platform key chain name',
+                'required' => true,
+                'cast' => 'string'
+            ],
+            'public_key_path' => [
+                'prefix' => 'pkp',
+                'longPrefix' => 'public_key_path',
+                'description' => 'Lti Platform public key path',
+                'required' => true,
+                'cast' => 'string'
+            ],
+            'private_key_path' => [
+                'prefix' => 'kp',
+                'longPrefix' => 'private_key_path',
+                'description' => 'Lti Platform private key path',
+                'required' => true,
+                'cast' => 'string'
+            ],
+            'private_key_passphrase' => [
+                'prefix' => 'kpp',
+                'longPrefix' => 'private_key_passphrase',
+                'description' => 'Lti Platform private key passphrase',
+                'required' => false,
+                'cast' => 'string'
+            ],
+        ];
+    }
+
+    protected function provideDescription()
+    {
+        return 'Script to create a LTI Platform Key Chain';
+    }
+
+    protected function provideUsage()
+    {
+        return [
+            'prefix' => 'h',
+            'longPrefix' => 'help',
+            'description' => 'Prints the help.'
+        ];
+    }
+
+    protected function run()
+    {
+        $keyId = $this->getOption('key_id');
+        $keyName = $this->getOption('key_name');
+        $publicKeyPath = $this->getOption('public_key_path');
+        $privateKeyPath = $this->getOption('private_key_path');
+        $privateKeyPassphrase = $this->getOption('private_key_passphrase');
+
+        if (empty($keyId) || empty($keyName) || empty($publicKeyPath) || empty($privateKeyPath)) {
+            return Report::createError(
+                'Not all required arguments were provided. Try to run the script with -h option'
+            );
+        }
+        $platformKeyChainRepository = $this->getServiceLocator()->get(PlatformKeyChainRepository::SERVICE_ID);
+        $options = $platformKeyChainRepository->getOptions();
+        $option = [
+            PlatformKeyChainRepository::OPTION_DEFAULT_KEY_ID => $keyId,
+            PlatformKeyChainRepository::OPTION_DEFAULT_KEY_NAME => $keyName,
+            PlatformKeyChainRepository::OPTION_DEFAULT_PUBLIC_KEY_PATH => $publicKeyPath,
+            PlatformKeyChainRepository::OPTION_DEFAULT_PRIVATE_KEY_PATH => $privateKeyPath,
+        ];
+
+        if (!empty($privateKeyPassphrase)) {
+            $option[PlatformKeyChainRepository::OPTION_DEFAULT_PRIVATE_KEY_PASSPHRASE] = $privateKeyPassphrase;
```

Comment:
**praise**: Thank you so much for this work, this small improvement can give us a big security boost going forward!